File: CS/DXApplication1/Form1.cs
using DevExpress.XtraBars.Ribbon;
using DevExpress.Spreadsheet;

namespace DXApplication1 {
    public partial class Form1 : RibbonForm {

        nwindDataSet dataSet;
        nwindDataSetTableAdapters.InvoicesTableAdapter adapter;
        IWorkbook template;

        public Form1() {
            InitializeComponent();

            dataSet = new nwindDataSet();
            adapter = new nwindDataSetTableAdapters.InvoicesTableAdapter();
            adapter.Fill(dataSet.Invoices);

            template = spreadsheetControl1.Document;

            template.MailMergeDataSource = dataSet;
            template.MailMergeDataMember = "Invoices";
        }

        private void Form1_Load(object sender, System.EventArgs e) {
            spreadsheetControl1.LoadDocument("Documents\\MailMergeTemplate_GroupData.xlsx");
        }
    }
}

File: VB/DXApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.Spreadsheet

Partial Public Class Form1
    Inherits RibbonForm

    Private dataSet As nwindDataSet
    Private adapter As nwindDataSetTableAdapters.InvoicesTableAdapter
    Private template As IWorkbook

	Public Sub New()
        InitializeComponent()

        dataSet = New nwindDataSet()
        adapter = New nwindDataSetTableAdapters.InvoicesTableAdapter()
        adapter.Fill(dataSet.Invoices)

        template = SpreadsheetControl1.Document

        template.MailMergeDataSource = dataSet
        template.MailMergeDataMember = "Invoices"
	End Sub

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        SpreadsheetControl1.LoadDocument("Documents\MailMergeTemplate_GroupData.xlsx")
    End Sub
End Class
